depdive/src/super_toml.rs

```diff
@@ -112,6 +113,42 @@ impl SuperPackageGenerator {
     }
 }
 
+pub struct TomlChecker;
+
+#[derive(Debug, Clone, PartialEq)]
+pub enum TomlType {
+    Package,
+    Workspace,
+}
+
+impl TomlChecker {
+    fn is_package_toml(path: &Utf8Path) -> Result<bool> {
+        let toml = read_to_string(path)?;
+        Ok(toml.trim_start().starts_with("[package]"))
+    }
+
+    fn is_workspace_toml(path: &Utf8Path) -> Result<bool> {
+        let toml = read_to_string(path)?;
+        Ok(toml.trim_start().starts_with("[workspace]"))
+    }
```

Comment:
This isn't sufficient since those sections can appear anywhere in a toml file. Instead you should probably parse the toml file with https://docs.rs/toml/0.5.8/toml/ and then query for the package or workspace sections.

Reply:
This was wrong naming. I intended to mean virtual manifests which is corrected now. Can it be possible to have other sections before `workspace` in case of a virtual manifest ?